src/Microsoft.Bcl.HashCode/src/Microsoft.Bcl.HashCode.csproj

```diff
@@ -9,18 +9,17 @@
     <VersionPrefix>1.1.1</VersionPrefix>
     <VersionPrefix Condition="'$(IsPackable)' == 'true'">1.2.0</VersionPrefix>
     <PackageValidationBaselineVersion>1.1.1</PackageValidationBaselineVersion>
+    <IsNetStandard21 Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))">true</IsNetStandard21>
   </PropertyGroup>
 
-  <PropertyGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))">
+  <PropertyGroup Condition="'$(IsNetStandard21)' == 'true'">
     <EnableDefaultItems>false</EnableDefaultItems>
   </PropertyGroup>
 
   <ItemGroup>
-    <Compile Include="TypeForwards.cs"
-             Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))" />
-    <Compile Remove="TypeForwards.cs"
-             Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))" />
-    <Compile Include="$(CommonPath)System\LocalAppContextSwitches.Common.cs" Link="Common\System\LocalAppContextSwitches.Common.cs" />
+    <Compile Include="TypeForwards.cs" Condition="'$(IsNetStandard21)' == 'true'" />
+    <Compile Remove="TypeForwards.cs"  Condition="'$(IsNetStandard21)' != 'true'" />
+    <Compile Include="$(CommonPath)src\System\LocalAppContextSwitches.Common.cs" Link="Common\System\LocalAppContextSwitches.Common.cs" />
```

Comment:
Looks like this was already wrong in main. We shouldn't compile this file for the full facade assembly: netstandard2.1

Reply:
```suggestion
```

Reply:
Did you mean to suggest the removal of line 19 instead, the one without the `!` negation?

I removed line 24 as suggested and got this error:

```
D:\maintenance-packages\src\Microsoft.Bcl.HashCode\src\LocalAppContextSwitches.cs(14,20): error CS0103: The name 'GetCachedSwitchValue' does not exist in the current context [D:\maintenance-packages\src\Microsoft 
.Bcl.HashCode\src\Microsoft.Bcl.HashCode.csproj::TargetFramework=netstandard2.0]
```

But if I remove line 19, the build succeeds.

Reply:
What about line 20? TypeForwards.cs. If I remove it, the build also passes. Which makes me wonder if the condition is needed at all.

Reply:
L22 should be conditioned on `Condition="'$(IsNetStandard21)' != 'true'"`

Reply:
I'm still not sure what you're asking me to do. This is the code we have:

- The first group has a `Compile Remove` and a `Compile Include`, in that order:
```csproj
18  <ItemGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))">
19    <Compile Remove="$(CommonPath)src\System\LocalAppContextSwitches.Common.cs" />
20    <Compile Include="TypeForwards.cs" />
21  </ItemGroup>
```

The second group has a `Compile Include` and a `Compile Remove`, in that order:
```csproj
23  <ItemGroup Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))">
24    <Compile Include="$(CommonPath)src\System\LocalAppContextSwitches.Common.cs" Link="Common\System\LocalAppContextSwitches.Common.cs" />
25    <Compile Remove="TypeForwards.cs" />
26  </ItemGroup>
```

The code compiles as it is. If I apply your suggestion (to remove the include), the build fails, because now the GetCachedSwitchValue API cannot be found in neither net462 or netstandard2.0:

```
D:\maintenance-packages\src\Microsoft.Bcl.HashCode\src\LocalAppContextSwitches.cs(14,20): error CS0103:
The name 'GetCachedSwitchValue' does not exist in the current context
[D:\maintenance-packages\src\Microsoft
.Bcl.HashCode\src\Microsoft.Bcl.HashCode.csproj::TargetFramework=net462]
```
```
D:\maintenance-packages\src\Microsoft.Bcl.HashCode\src\LocalAppContextSwitches.cs(14,20): error CS0103:
The name 'GetCachedSwitchValue' does not exist in the current context
[D:\maintenance-packages\src\Microsoft 
.Bcl.HashCode\src\Microsoft.Bcl.HashCode.csproj::TargetFramework=netstandard2.0]
```

Reply:
Looks like the line that we are commenting is stale. Sorry, my bad. Now it's this line: https://github.com/carlossanlop/maintenance-packages/blob/523240065ac2ce62322423842ad67392e58a1273/src/Microsoft.Bcl.HashCode/src/Microsoft.Bcl.HashCode.csproj#L19

You don't need it as the file isn't auto-included because `EnableDefaultItems` is set to false for netstandard2.1 and NetMinimum.